### PR TITLE
Support glm 4v

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

support [THUDM/glm-4v-9b](https://huggingface.co/THUDM/glm-4v-9b) for pytorch engine

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: lmdeploy/pytorch/config.py [hunk 1]
```diff
@@ -84,6 +84,7 @@ class ModelConfig:
     model_arch: str = None
     unused_modules: List[str] = None
     auto_model_cls: Any = AutoModelForCausalLM
+    cogvlm_style: bool = False
```
Comment (grimoire) [inline lmdeploy/pytorch/config.py:87]: Do we have any better name?

File: lmdeploy/pytorch/models/chatglm2.py [hunk 2]
```diff
@@ -268,6 +282,21 @@ def _contiguous_batching_forward(
             past_key_values: Optional[Tuple[Tuple[torch.Tensor, torch.Tensor],
                                             ...]] = None,
             inputs_embeds: Optional[torch.Tensor] = None):
+        assert input_ids is not None
+        context = self.context.context
+        # get inputs from context
+        vision_embeddings = context.input_embeddings
+        vision_embedding_indexing = context.input_embedding_indexing
+
+        # chatglm2/chaglm3 uses hidden_states in shape of [sq,b,h]
```
Comment (grimoire) [inline lmdeploy/pytorch/models/chatglm2.py:291]: This is so sick :-(

**Comment by grimoire** [comment]
After comparation between https://huggingface.co/THUDM/glm-4v-9b/blob/main/modeling_chatglm.py and https://huggingface.co/THUDM/chatglm3-6b/blob/main/modeling_chatglm.py , I found that almost all  `__init__` are the same. In theory we can share same implementation without the `_is_chatglm` flag.

File: lmdeploy/pytorch/config.py (hunk 1 above)
Comment (RunningLeon) [reply lmdeploy/pytorch/config.py:87]: don't know, you offer one?

File: lmdeploy/pytorch/models/chatglm2.py (hunk 2 above)
Comment (RunningLeon) [reply lmdeploy/pytorch/models/chatglm2.py:291]: sadly there's little we can do about it

File: tests/test_lmdeploy/test_model.py [hunk 3]
```diff
@@ -267,6 +267,32 @@ def test_deepseek_coder():
     assert res.startswith(ref)
 
 
+def test_chatglm3():
+    model_path_and_name = 'THUDM/chatglm3-6b'
```
Comment (AllentDan) [inline tests/test_lmdeploy/test_model.py:271]: why not use glm4v?

Comment (RunningLeon) [reply tests/test_lmdeploy/test_model.py:271]: glm4v has beed added in line 296

**Review by grimoire — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**